lib/irb.rb

```diff
@@ -1222,6 +1222,8 @@ def handle_exception(exc)
         irb_bug = true
       else
         irb_bug = false
+        backtrace = exc.backtrace.map { |l| @context.workspace.filter_backtrace(l) }.compact
+        exc.set_backtrace(backtrace)
```

Comment:
That's a great point, I'll push up that change

Reply:
Hmm, this results in some test failures, whether added directly above `exc.set_backtrace(backtrace)` or at the very top of the method (before the conditional). I wonder if its because `exc.dup` creates a shallow copy?

I will investigate those failures more to see if `exc.dup` can work, but if not, should I look for some other way to avoid modifying the original exception's backtrace?

Reply:
Sounds good. I appreciate the feedback, I'll explore the options when I have some time.

Should I close this PR and open a new one when I have a solution that doesn't modify the original exception's backtrace? Or should I just push that solution here when it is ready?

Reply:
Actually, I tried using `exc.clone` rather than `exc.dup` and that appears to resolve whatever part of the state wasn't getting copied using `dup`.